### ISSUE TITLE
Publishing an exe as "PublishSingleFile=true" with version 2.3.4 fails

### ISSUE THREAD

**Issue by C302** [Tyrrrz/CliFx#137]
### Version

2.3.4

### Platform

win-x64

### Steps to reproduce

publishing an exe cli as "-r win-x64 -p:PublishSingleFile=true --self-contained true" with 2.3.3 works, but fails with 2.3.4

### Details

![image](https://github.com/Tyrrrz/CliFx/assets/78357766/d99b6317-8901-49ad-9a3f-7684bb083ae7)

### Checklist

- [X] I have looked through existing issues to make sure that this bug has not been reported before
- [X] I have provided a descriptive title for this issue
- [X] I have made sure that that this bug is reproducible on the latest version of the package
- [X] I have provided all the information needed to reproduce this bug as efficiently as possible
- [ ] I have sponsored this project

**Comment by Storm-BE** [comment]
The error above is only output when running the publish output, not during publishing itself.

**Comment by Tyrrrz** [comment]
Yes, I removed the fallback because I wanted the user to be aware that the name could not be resolved. In order to specify the application name explicitly use the methods on the `CliApplicationBuilder` instance.

Unless I'm missing something and it's actually possible to retrieve the entry assembly when publishing a self-contained binary, I'm going to close this as it's not actually a bug.
